sdk/src/main/java/com/mapbox/maps/renderer/MapboxWidgetRenderer.kt

```diff
@@ -0,0 +1,204 @@
+
+  fun getTexture() = textures[0]
+
+  fun onSharedContext(sharedContext: EGLContext) {
+    if (eglPrepared) {
+      release()
+    }
+    eglCore = EGLCore(
+      translucentSurface = false,
+      antialiasingSampleCount = antialiasingSampleCount,
+      sharedContext = sharedContext,
+    )
+  }
+
+  fun onSurfaceChanged(width: Int, height: Int) {
+    sizeChanged = true
+    this.width = width
+    this.height = height
+    widgets.forEach { it.renderer.onSurfaceChanged(width, height) }
+  }
+
+  private fun attachTexture() {
+    if (textures[0] != 0) {
+      GLES20.glDeleteTextures(textures.size, textures, 0)
+    }
+    GLES20.glGenTextures(1, textures, 0)
+
+    GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, textures[0])
+    GLES20.glTexParameterf(
+      GLES20.GL_TEXTURE_2D,
+      GLES20.GL_TEXTURE_MAG_FILTER,
+      GLES20.GL_LINEAR.toFloat()
+    )
+    GLES20.glTexParameterf(
+      GLES20.GL_TEXTURE_2D,
+      GLES20.GL_TEXTURE_MIN_FILTER,
+      GLES20.GL_LINEAR.toFloat()
+    )
+    GLES20.glTexParameterf(
+      GLES20.GL_TEXTURE_2D,
+      GLES20.GL_TEXTURE_WRAP_S,
+      GLES20.GL_CLAMP_TO_EDGE.toFloat()
+    )
+    GLES20.glTexParameterf(
+      GLES20.GL_TEXTURE_2D,
+      GLES20.GL_TEXTURE_WRAP_T,
+      GLES20.GL_CLAMP_TO_EDGE.toFloat()
+    )
+    GLES20.glTexImage2D(
+      GLES20.GL_TEXTURE_2D,
+      0,
+      GLES20.GL_RGBA,
+      width,
+      height,
+      0,
+      GLES20.GL_RGBA,
+      GLES20.GL_UNSIGNED_BYTE,
+      null
+    )
+    GLES20.glFramebufferTexture2D(
+      GLES20.GL_FRAMEBUFFER, GLES20.GL_COLOR_ATTACHMENT0,
+      GLES20.GL_TEXTURE_2D, textures[0], 0
+    )
+    GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, 0)
+  }
+
+  fun release() {
+    val eglCore = this.eglCore
+    val eglSurface = this.eglSurface
+    if (eglCore != null) {
+      if (eglSurface != null && eglSurface != eglCore.eglNoSurface) {
+        eglCore.makeCurrent(eglSurface)
+
+        GLES20.glDeleteFramebuffers(framebuffers.size, framebuffers, 0)
+        GLES20.glDeleteTextures(textures.size, textures, 0)
+        widgets.forEach {
+          it.renderer.release()
+        }
+        widgets.clear()
+
+        eglCore.releaseSurface(eglSurface)
+      }
+
+      eglCore.release()
+    }
+    this.eglSurface = null
+    this.eglCore = null
+  }
+
+  fun updateTexture() {
+    if (needRender) {
+      checkSizeChanged()
+      checkEgl()
+      val eglCore = this.eglCore
+      val eglSurface = this.eglSurface
+      if (eglCore != null && eglSurface != null && eglSurface != eglCore.eglNoSurface) {
+        eglCore.makeCurrent(eglSurface)
+        bindFramebuffer()
+        attachTexture()
+        widgets.forEach {
+          it.renderer.render()
+        }
+        unbindFramebuffer()
+      }
+    }
+  }
+
+  private fun checkSizeChanged() {
+    if (sizeChanged) {
+      val eglCore = this.eglCore
+      val eglSurface = this.eglSurface
+      if (eglCore != null && eglSurface != null && eglSurface != eglCore.eglNoSurface) {
```

Comment:
this check is used plenty of times - perhaps add single bool var with an explicit getter?